pydicom/filereader.py

```diff
@@ -42,6 +42,59 @@
 sys_is_little_endian = (byteorder == 'little')
 
 
+def _auto_open(filepath, *args, **kwargs):
+    """Auto-magically (transparently) open a compressed file.
+
+    Supports `gzip` and `bzip2`.
+
+    Note: all compressed files should be opened as binary.
+    Opening in text mode is not supported.
+
+    Parameters
+    ----------
+    filepath : str
+        The file path.
+    *args : iterable
+        Positional arguments passed to `open()`.
+    **kwargs : dict
+        Keyword arguments passed to `open()`.
+
+    Returns
+    -------
+    file
+        A file object.
+
+    Raises
+    ------
+    IOError
+        on failure.
+
+    See Also
+    --------
+    open(), gzip.open(), bz2.open()
+    """
+    try:
```

Comment:
To save on the import, e.g. if you have a `gz` file, and you load it, there is no need to import `bz2`.